docs: improve reference page on Ubuntu distributions (#167)
canonical/open-documentation-academy: wsl/reference/distributions.md

# Distributions

> Our flagship distribution (distro) is Ubuntu. It is the default option when you install WSL for the first time. However, we develop several flavours for the Ubuntu distro and one or more of these flavours may fit your needs better.

Author: Noted. Thanks for the suggestion

Each of these flavours corresponds to a different application on the Microsoft Store, and once installed, they'll create different flavors in your WSL. 

## Ubuntu Variants
These are the variants of the default Ubuntu flavor we develop and maintain:

- [Ubuntu](https://apps.microsoft.com/detail/9PDXGNCFSCZV?hl=en-us&gl=US) ships the latest stable LTS release of Ubuntu. When new LTS versions are released, Ubuntu can be upgraded once the first point release is available.
- Numbered releases, for example [Ubuntu 22.04.3 LTS](https://apps.microsoft.com/detail/9PN20MSR04DW?hl=en-us&gl=US) refer to specific Long Term Stability (LTS) releases that are supported for five years. For more information on releases, support and timelines, visit the [Ubuntu releases page](https://wiki.ubuntu.com/Releases).

> - [Ubuntu (Preview)](https://apps.microsoft.com/detail/9P7BDVKVNXZ6?hl=en-us&gl=US) is a daily build of the latest development version of Ubuntu previewing new features as they are developed. It does not receive the same level of QA as stable releases and should not be used for production workloads.

Reviewer: At the end of this section, we could include an admonition/comment, notifying the reader that:

> Interim releases of Ubuntu are currently not supported on WSL.

## Naming

> The variants of the default Ubuntu flavor is available under different names depending the context you try to access it with. Here is a table matching them.

Reviewer: ```suggestion
Depending on context, releases of Ubuntu are referred to by different names:
```

> 1. App name is the name you'll see in the Microsoft Store and Winget.

Reviewer: In this and others below, I tried to add more context where possible.

2. AppxPackage is the name you'll see in `Get-AppxPackage`.

> 3. Distro name is the name you'll see when doing `wsl -l -v` or `wsl -l --online`.

Reviewer: ```suggestion
3. **Distro name** is the name logged when you invoke `wsl -l -v` to list installed releases of Ubuntu.
```

4. Executable is the program you need to run to start the distro.

> 

Reviewer: These naming conventions are summarised in the table below:

> | App name             | AppxPackage name                       | Variant      | Executable          |

Reviewer: ```suggestion
| App name             | AppxPackage name                       | Distro name     | Executable          |
```

Let's keep this as-is.

Author: Any particular reason why? I don't think "Distro" accurately describes it.

Reviewer: While I understand the logic behind the alternative word choice, I do not find sufficient justification to introduce new terminology that could be unconventional or inconsistent with that found in other documentation.

For example, in documentation for the `wsl -l` command in https://learn.microsoft.com/en-us/windows/wsl/basic-commands#list-available-linux-distributions , the items listed by the command are referred to as "distributions".

In addition, I do think it is reasonably accurate. Perhaps it is slightly more indirect for Ubuntu and Ubuntu (Preview), which points to the latest and development releases for the distribution, but I don't think we should worry about that too much for now.

| -------------------- | -------------------------------------- | ---------------- | ------------------- |
| `Ubuntu`             | `CanonicalGroupLimited.Ubuntu`         | `Ubuntu`         | `ubuntu.exe`        |

> | `Ubuntu (Preview)`   | `CanonicalGroupLimited.UbuntuPreview`  | `Ubuntu-Preview` | `ubuntupreview.exe` |

Reviewer: I suggest we add a note on the WSL kernel.
The reason is that we are talking about Ubuntu releases and support, but we do not actually maintain the kernel for WSL.
By having no mention of this here, it is difficult to blame someone who assumes that we maintain the kernel, and they may be disappointed when we can't provide support

It could simply be an admonition/note at the very end to the effect:

> Note:
> The kernel used in WSL environments is maintained by Microsoft.
> Bug reports and support requests for the WSL kernel should be direct to the [official repository](https://github.com/microsoft/WSL2-Linux-Kernel) for the WSL kernel.

Let me know if you have any thoughts on this.

Author: Personally, when I was going through the doc, there was nothing that suggested that the Kernel was maintained by Canonical (and it seems like common knowledge that the kernel is maintained by Microsoft). But you make a valid point, we need to make sure we leave nothing to assumption. I will add your suggested note.

Reviewer: Thanks. I understand what you mean.

It's not common knowledge by the way.

We have had some users asking about this.

Depending on context, releases of Ubuntu are referred to by different names: 

1. **App name** is the name of the application for a specific Ubuntu release that you will see in the Microsoft Store.  
2. **AppxPackage** is the name that can be passed to `Get-AppxPackage -Name` in PowerShell to get information about an installed package. 
3. **Distro name** is the name logged when you invoke `wsl -l -v` to list installed releases of Ubuntu.  
4. **Executable** is the program you need to run to start the Ubuntu distro.

These naming conventions are summarised in the table below:

| App name             | AppxPackage name                       | Distro name      | Executable          |
| -------------------- | -------------------------------------- | ---------------- | ------------------- |
| `Ubuntu`             | `CanonicalGroupLimited.Ubuntu`         | `Ubuntu`         | `ubuntu.exe`        |
| `Ubuntu (Preview)`   | `CanonicalGroupLimited.UbuntuPreview`  | `Ubuntu-Preview` | `ubuntupreview.exe` |
| `Ubuntu XX.YY.Z LTS` | `CanonicalGroupLimited.UbuntuXX.YYLTS` | `Ubuntu-XX.YY`   | `ubuntuXXYY.exe`    |

:::{note}
The kernel used in WSL environments is maintained by Microsoft.

> Bug reports and support requests for the WSL kernel should be direct to the official repository for the WSL kernel.

Reviewer: ```suggestion
Bug reports and support requests for the WSL kernel should be directed to the [official repository for the WSL kernel](https://github.com/microsoft/WSL2-Linux-Kernel).
```
